plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -80,192 +197,724 @@ function executeCLI($scriptName, &$args) {
+		$submissionFileDao = $this->getCachedDao('SubmissionFileDAO');
+
+		/** @var SubmissionFile $submissionFile */
+		$submissionFile = $submissionFileDao->newDataObject();
+		$submissionFile->setData('submissionId', $submissionId);
+		$submissionFile->setData('uploaderUserId', $this->_userId);
+		$submissionFile->setSubmissionLocale($data->locale);
+		$submissionFile->setGenreId($genreId);
+		$submissionFile->setFileStage(SUBMISSION_FILE_PROOF);
+		$submissionFile->setAssocType(ASSOC_TYPE_REPRESENTATION);
+		$submissionFile->setData('assocId', $publicationFormatId);
+		$submissionFile->setData('createdAt', Core::getCurrentDate());
+		$submissionFile->setDateModified(Core::getCurrentDate());
+		$submissionFile->setData('mimetype', $mimeType);
+		$submissionFile->setData('fileId', $fileId);
+		$submissionFile->setData('name', pathinfo($filePath, PATHINFO_FILENAME));
+
+		// Assume open access, no price.
+		$submissionFile->setDirectSalesPrice(0);
+		$submissionFile->setSalesType('openAccess');
+
+		$submissionFileDao->insertObject($submissionFile);
+	}
+
+	/**
+	 * Process data for Keywords
+	 *
+	 * @param object $data
+	 * @param int $publicationId
+	 */
+	private function processKeywords($data, $publicationId) {
+		$keywordsList = [$data->locale => explode(';', $data->keywords)];
+
+		if (count($keywordsList[$data->locale]) > 0) {
+			/** @var SubmissionKeywordDAO $submissionKeywordDao */
+			$submissionKeywordDao = $this->getCachedDao('SubmissionKeywordDAO');
+			$submissionKeywordDao->insertKeywords($keywordsList, $publicationId);
+		}
 	}
-}
 
+	/**
+	 * Process data for Subjects
+	 *
+	 * @param object $data
+	 * @param int $publicationId
+	 */
+	private function processSubjects($data, $publicationId) {
+		$subjectsList = [$data->locale => explode(';', $data->subjects)];
+
+		if (count($subjectsList[$data->locale]) > 0) {
+			/** @var SubmissionSubjectDAO $submissionSubjectDao */
+			$submissionSubjectDao = $this->getCachedDao('SubmissionSubjectDAO');
+			$submissionSubjectDao->insertSubjects($subjectsList, $publicationId);
+		}
+	}
+
+	/**
+	 * Process data for the Book Cover Image
+	 *
+	 * @param object $data
+	 * @param string $srcFilePath
+	 * @param Publication $publication
+	 */
+	private function processBookCoverImage($data, $srcFilePath, $publication) {
+		$press = $this->_presses[$data->pressPath];
+		$coverImagelocale = [];
+		$coverImage = [];
 
+		$sanitizedCoverImageName = str_replace([' ', '_', ':'], '-', mb_strtolower($data->bookCoverImage));
+		$sanitizedCoverImageName = PKPstring::regexp_replace('/[^a-z0-9\.\-]+/', '', $sanitizedCoverImageName);
+		$sanitizedCoverImageName = basename($sanitizedCoverImageName);
+
+		$coverImage['uploadName'] = uniqid() . '-' . $sanitizedCoverImageName;
+		$coverImage['altText'] = $bookCoverImageAltText ?? '';
+
+		$destFilePath = $this->_publicFileManager->getContextFilesPath($press->getId()) . '/' . $coverImage['uploadName'];
+		copy($srcFilePath, $destFilePath);
```

Comment:
It might be interesting to check the return of the `copy()` and perhaps other places that might fail.

We're not using transactions nor checking if the database operations worked fine, so such things don't need to be checked 🙈

Reply:
Second comment... I think it's better to use the `FileManager::copy()` to setup the expected permissions.